src/language-js/embed.js

```diff
@@ -286,29 +286,39 @@ function replacePlaceholders(quasisDoc, expressionDocs) {
     }
     const atPlaceholderIndex = parts.findIndex(
       (part) =>
-        typeof part === "string" && part.startsWith("@prettier-placeholder")
+        typeof part === "string" && part.includes("@prettier-placeholder")
     );
     if (atPlaceholderIndex > -1) {
-      const placeholder = parts[atPlaceholderIndex];
-      const rest = parts.slice(atPlaceholderIndex + 1);
-      const placeholderMatch = placeholder.match(
-        /@prettier-placeholder-(.+)-id([\S\s]*)/
-      );
-      const placeholderID = placeholderMatch[1];
-      // When the expression has a suffix appended, like:
-      // animation: linear ${time}s ease-out;
-      const suffix = placeholderMatch[2];
-      const expression = expressions[placeholderID];
+      const part = parts[atPlaceholderIndex];
+      // When we have multiple placeholders in one line, like:
+      // ${Child}${Child2}:not(:first-child)
+      const placeholders = part.split("@prettier-placeholder");
+      const tokens = [];
+      placeholders.forEach((placeholder, idx) => {
+        if (idx === 0) {
+          // The first item is either empty or the prefix not an actual placeholder
+          tokens.push(placeholder);
+          return;
+        }
 
-      replaceCounter++;
-      parts = parts
-        .slice(0, atPlaceholderIndex)
-        .concat(["${", expression, "}" + suffix])
-        .concat(rest);
+        placeholder = "@prettier-placeholder" + placeholder;
+        const placeholderMatch = placeholder.match(
+          /@prettier-placeholder-(.+)-id([\S\s]*)/
```

Comment:
Please make this regexp stricter to avoid problems like the following:

**Prettier pr-8097**
[Playground link](https://deploy-preview-8097--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBnGBPANnAJgOlwEsA3AAwBJgAzCCAXwFojccACYN+skAGhAgAHGEWipkoAIYAnaRADuABRkJxKSVnmSM4-gCNpksAGs4MAMqCjRKAHNkMaQFc4-OAFs9eXHgAykuydJWzgAMQhpd0kYETtkEEknGAg+EAALGHcsAHU0onhUKzA4c1V80nyMeLBUXRAbVDhpGEVDWyjkag1G-gArVAAPACFDEzNzSXc4Xxs4Tu7XEH6B8xtbHABFJwh4eawekCtpRul4vUkvLFTBaRsYbJYYNOQADgAGfhuIRuzDQXibnATiQ5vwAI7beCtIRqBKoRhQOB4PCpaRwCFENGtYIdJBdfaLRruIgOZyEtabSFzPELfgwC4PXBPZAAJjphiIWDWAGEIO5cSAgQBWVJORoAFQuanxBxILgAklAfLBzGBbsIAIJK8yYHB7Rr0ehAA)
```sh
--parser babel
```

**Input:**
```jsx
styled.div`${foo}-idle { }`
```

**Output:**
```jsx
TypeError: Cannot read property 'type' of undefined
    at propagateBreaksOnEnterFn (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:15880:15)
    at traverseDoc (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:15740:13)
    at Object.propagateBreaks (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:15905:5)
    at printAstToDoc (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:16940:16)
    at coreFormat (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:17200:17)
    at format (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:17420:77)
    at formatWithCursor (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:17436:14)
    at https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:33586:14
    at Object.format (https://deploy-preview-8097--prettier.netlify.app/lib/standalone.js:33595:14)
    at formatCode (https://deploy-preview-8097--prettier.netlify.app/worker.js:233:21)
```

Reply:
Fixed both of them.